IssueTracker/IssueTrackerAPI/Controllers/ParticipantController.cs

```diff
@@ -59,12 +62,68 @@ public async Task<IResult> GetParticipant(int id)
             return Results.Ok(participantResponse);
 
         }
+        [HttpPost("createowner")]
+        [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme)]
+        public async Task<IResult> CreateOwner([FromBody]int projectId)
+        {
+            if (projectId > 0)
+            {
+                
+                var idclaim = User.Claims.FirstOrDefault(x => x.Type.Equals("UserId"));
+                if (idclaim != null)
+                {
+                    bool result = await CheckRole.OwnerExists(_participantData, projectId);
+                    if (result == false)
+                    {
+                        Participant participant = new Participant()
+                        {
+                            ProjectId = projectId,
+                            RoleId = Roles.Owner,
+                            UserId = Guid.Parse(idclaim.Value)
+
+                        };
+                        await _participantData.AddAsync(participant);
+                        return Results.Ok();
+                    }
+                }
+            }
+            return Results.BadRequest();
+        }
         [HttpPost("create")]
+        [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme)]
         public async Task<IResult> CreateParticipant(ParticipantRequest participantRequest)
         {
-            var participant = mapper.Map<Participant>(participantRequest);
-            await _participantData.AddAsync(participant);
-            return Results.Ok();
+            if(!ParticipantRequestValidation.IsValid(participantRequest)) return Results.BadRequest();
+            var userclaim = User.Claims.FirstOrDefault(x => x.Type.Equals(ClaimTypes.Name));
+            var emailclaim = User.Claims.FirstOrDefault(x => x.Type.Equals(ClaimTypes.Email));
+            var idclaim = User.Claims.FirstOrDefault(x => x.Type.Equals("UserId"));
+            if (userclaim != null && emailclaim != null && idclaim != null)
```

Comment:
1. why are you checking all 3 claim? the id should be enough, that's the only one used below
2. you should return the new Id value in a create method. The client might want to take action with the new entity